### PR TITLE
[Doc] add-faq-for-real-time-synchronization-from-mysql-into-sr

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Synchronize data from MySQL in real time
+
+## What do I do if a Flink job reports an error?
+
+A Flink job reports the error `Could not execute SQL statement. Reason:org.apache.flink.table.api.ValidationException: One or more required options are missing.`
+
+It is possible that the required configuration information is missing in multiple sets of rules, such as `[table-rule.1]` and `[table-rule.2]`, in the SMT configuration file **config_prod.conf**.
+
+You can check whether each set of rules, such as `[table-rule.1]` and `[table-rule.2]`. is configured with the required database, table, and Flink connector information.
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:9]: image
多余逗号

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Synchronize data from MySQL in real time
+
+## What do I do if a Flink job reports an error?
+
+A Flink job reports the error `Could not execute SQL statement. Reason:org.apache.flink.table.api.ValidationException: One or more required options are missing.`
+
+It is possible that the required configuration information is missing in multiple sets of rules, such as `[table-rule.1]` and `[table-rule.2]`, in the SMT configuration file **config_prod.conf**.
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:7]: image
-> A possible reason is that ...?

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+execution.checkpointing.interval: 300000
+state.backend: filesystem
+state.checkpoints.dir: file:///tmp/flink-checkpoints-directory
+```
+
+Parameter description:
+
+> **NOTE**
+>
+> For more detailed parameter descriptions in Flink documentation,  see [Checkpointing](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/fault-tolerance/checkpointing/).
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:27]: image
有多余空格

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+- `execution.checkpointing.interval`: the base time interval of checkpointing. Unit: millisecond. To enable the checkpointing mechanism, you need to set this parameter to a value greater than `0`.
+- `state.backend`: How the state is represented internally, and how and where it is persisted upon checkpoints depends on the chosen **State Backend**. Common values are `filesystem` or `rocksdb`. After the checkpointing mechanism is enabled, the state is persisted upon checkpoints to prevent data loss and ensure data consistency after recovery.  For more information on state, see [State Backends](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/state/state_backends/).
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:30]: image
文风一致性。改为 -> specifies how the state is represented internally, and how and where it is persisted upon checkpointing.

Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:30]: image
有多余空格

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+- `state.checkpoints.dir`: the directory to which checkpoints are written to.
+
+## How can I manually stop a Flink job and later restore it to the state before stopping?
+
+You can manually trigger a [savepoint](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/state/savepoints/) when stopping a Flink job (a savepoint is a consistent image of the execution state of a streaming Flink job, and is created based on the checkpointing mechanism). Later, you can restore the Flink job from the specified savepoint.
+
+1. Stop a Flink job with a savepoint. The following command automatically triggers a savepoint for the Flink job `jobId` and stops the Flink job. Additionally, you can specify a target file system directory to store the savepoint.
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:37]: image
这里是举例，一个步骤流程，应该是用 the 比较好

File: docs/faq/loading/synchronize_mysql_into_sr.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    Parameter description:
+
+    - `jobId`: You can view the Flink job ID from the Flink WebUI or by running `flink list -running` on the command line.
+    - `targetDirectory`: You can specify `state.savepoints.dir` as the default directory for storing savepoints in the Flink configuration file **flink-conf.yml**. When a savepoint is triggered, the savepoint is stored in this default directory and you do not need to specify a directory .
+
+    ```Bash
+    state.savepoints.dir: [file:// or hdfs://]/home/user/savepoints_dir
+    ```
+
+2. If you need to restore a Flink job to its state before it was stopped, specify the savepoint when resubmitting the Flink job.
```
Comment (amber-create) [inline docs/faq/loading/synchronize_mysql_into_sr.md:52]: image
这句当前感觉是一个可选操作，但是看这个 faq 的话，应该是必须的步骤之一（第一步是停止的时候打一个 savepoint，第二步就是重新提交作业并且在提交的时候指定这个 savepoint，这样就可以把作业恢复到停止前的所在 savepoint 时候的那个状态了）。建议改为 -> Resubmit the Flink job with the preceding savepoint specified.

**Review by amber-create — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0 branch-2.5 branch-2.4 branch-2.3

**Pull request merged**